src/app/utils/state.ts

```diff
@@ -28,21 +27,29 @@ class StateManager {
 
   public async init() {}
 
-  public async update() {
-    console.log("supabase url", process.env.SUPABASE_URL);
-    const status = await this.getLatestStatus();
-    console.log("tamagotchiStatus", status);
+  public async update(vectorSearchResult?: any[]) {
+    const statusData = await this.getLatestStatus();
+    const status = statusData.status;
+    const preferences = vectorSearchResult
+      ? vectorSearchResult
+          .map((item) => (item.content ? item.content : ""))
+          .join("\n")
+      : "No preferences";
+    const lastStatusTs = statusData!.updatedat;
     const age = status!.age ? status!.age + 1 : 1; // 1 tick older!
```

Comment:
**issue:** Inconsistent null handling in age calculation

The code uses a non-null assertion but then checks for truthiness. Consider using optional chaining or standardize the null handling approach.